Introduce IWritableStorage interface (#10582)
friendica/friendica: doc/AddonStorageBackend.md

Override the two necessary instances:
```php
use Friendica\Model\Storage\ISelectableStorage;

abstract class StorageTest 
{
	// returns an instance of your newly created storage class
	abstract protected function getInstance();

	// Assertion for the option array you return for your new StorageClass
	abstract protected function assertOption(ISelectableStorage $storage);
} 
```

## Exception handling

There are two intended types of exceptions for storages

### `StorageException`

> This is the common exception in case during the storage backend occurs any, unexpected errors.

Reviewer: ```suggestion
This is the common exception in case unexpected errors happen using the storage backend.
```

> If there's a predecessor to this execption (e.g. you catched a exception an re-throw this execption), you should add the predecessor for transparent reasons.

Reviewer: ```suggestion
If there's a predecessor to this exception (e.g. you caught an exception and are throwing this execption), you should add the predecessor for transparency reasons.
```

Example:

```php
use Friendica\Model\Storage\ISelectableStorage;

class ExampleStorage implements ISelectableStorage 
{
	public function get(string $reference) : string
	{
		try {
			throw new Exception('a real bad exception');
		} catch (Exception $exception) {
			throw new \Friendica\Model\Storage\StorageException(sprintf('The Example Storage throws an exception for reference %s', $reference), 500, $exception);
		}
	}
} 
```

> ### `ReferenceStorageExecption`

Reviewer: Please move this more specific exception first.

Author: done

This storage exception should be used in case the caller tries to use an invalid references.
This could happen in case the caller tries to delete or update an unknown reference.
The implementation of the storage backend must not ignore invalid references.

Avoid throwing the common `StorageExecption` instead of the `ReferenceStorageException` at this particular situation!

## Example

> Here an hypotetical addon which register an unuseful storage backend.

Reviewer: ```suggestion
Here an hypotetical addon which register a useless storage backend.
```
